[Yim Jaewon] iP

somekoreanboi opened the pull request:
# JWBOT

Here comes your time management savior, JWBOT bros! It's,

* English text based
* Super easy to learn and use
* Fast and efficient

All you need to do is,

1. download it from [this link](https://github.com/somekoreanboi/ip).
1. double-click and execute the program
1. add your tasks to the bot
1. enjoy the task management functions of the bot

By the way, the bot is totally **FREE!**

The bot has following features:

✅| Managing tasks (Todo, Deadline, Event)
⬜️| Reminders (Not updated yet)

The app is made in Java, so you can use this to practice java if you want! Here's the main method:

```
public static void main(String[] args) {
    new JWBot(TXT_PATH).run();
}
```

Commit:
Add Gradle support

somekoreanboi committed:
level1 incremented

somekoreanboi committed:
level 2 incremented

somekoreanboi committed:
level 3 incremented

somekoreanboi committed:
level 4 incremented

somekoreanboi committed:
simple test implemented

somekoreanboi committed:
level 5 incremented

somekoreanboi committed:
level 6 incremented

somekoreanboi committed:
simple javadoc added

somekoreanboi committed:
Add txt file read and write function

somekoreanboi committed:
Add level 8

somekoreanboi committed:
Merge branch 'branch-Level-8'

somekoreanboi committed:
Fix Deadline Localdate format bug

somekoreanboi committed:
Improve OOP

somekoreanboi committed:
packages organized

somekoreanboi committed:
Add Junit Tests

somekoreanboi committed:
Javadoc added

somekoreanboi committed:
Fix some coding standards

somekoreanboi committed:
Implement find function

somekoreanboi committed:
Commit for renaming

somekoreanboi committed:
Merge branch 'branch-A-JavaDoc'

somekoreanboi committed:
Merge level 9; conflicts fixed

somekoreanboi committed:
Change minor things

somekoreanboi committed:
Merge branch 'add-gradle-support'

somekoreanboi committed:
Test checkstyle

somekoreanboi committed:
Implement JavaFX Gui

Jay-luozhijie commented on src/main/java/jwbot/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    /**
+     * The method that executes the command
```
Probably it can start with a verb? like Executes

Jay-luozhijie commented on src/main/java/jwbot/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,45 @@
+     * @return the boolean false to not stop the bot
+     */
+    @Override
+    public boolean isExit() {
```
Good naming!

Jay-luozhijie commented on src/main/java/jwbot/command/AddTodoCommand.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    /**
+     * The method that executes the command
```
Maybe the description can end with a period?

Jay-luozhijie commented on src/main/java/jwbot/data/TaskList.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    /**
+     * Add the task to the list.
```
I think the verb "Add" can be changed to "Adds"

somekoreanboi committed:
Change minor

zhangchengchuan commented on src/main/java/jwbot/command/AddEventCommand.java:
```diff
@@ -0,0 +1,45 @@
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws JwBotException {
+        try {
+            String content = input.split(" ", 2)[1];
+            String[] separated = content.split(" /at ");
```
Nice that you followed the coding standard and used String[] separated instead of String separated[] ! 👍

zhangchengchuan commented on src/main/java/jwbot/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,45 @@
+            tasks.addTask(deadline);
+            storage.write(tasks);
+            return ui.showAddTaskSuccessMessage(deadline);
+        } catch (Exception e) {
```
Is it possible for the exception to be more specific? 🤔

zhangchengchuan commented on src/main/java/jwbot/data/task/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+public class Deadline extends Task {
+
+    /**
+     * the date or time of the event.
```
Would it be neater if the first letter is capitalized? 😄

zhangchengchuan commented on src/main/java/jwbot/parser/Parser.java:
```diff
@@ -0,0 +1,50 @@
+     * @param input the command entered by the user
+     * @return a parsed Command object created based on the user command
+     */
+    public static Command parse(String input) throws JwBotException {
```
One suggestion, perhaps you could try out switch statements! 😀

zhangchengchuan commented on src/main/java/jwbot/storage/Storage.java:
```diff
@@ -0,0 +1,110 @@
+     *
+     * @param items the list of the tasks
+     */
+    public void write(TaskList items) throws IOException {
```
Nicely done! 👍

zhangchengchuan reviewed (commented):
LGTM. Just some nits to fix!

somekoreanboi committed:
Add some important assertions

There are some assertions that are to be made for the bot to run.
Such assertions listed below should be kept.

Let's,
* assert all commands are executed with adequate input
* assert none of the parameters of necessary methods are null

somekoreanboi committed:
Improve CodeQuality: make the code abstraction level same

Some codes in Command classes have different level of abstraction.

Some codes could be seperated into private methods for improvement.

Hence, let's add a protected processtask method for all command classes.

somekoreanboi committed:
Merge pull request #2 from somekoreanboi/branch-A-Assertions

Add some important assertions

somekoreanboi committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	jwbot.txt

somekoreanboi committed:
Implement undo function

somekoreanboi committed:
Fix undo functions bugs

somekoreanboi committed:
Add user-guide

somekoreanboi committed:
Release final submission jar
